dataflow/src/org/checkerframework/dataflow/analysis/Store.java

```diff
@@ -54,6 +54,25 @@
     S leastUpperBound(S other);
 
     /**
+     * Compute an upper bound of two stores that is wider than the least upper bound of the two
+     * stores.
```

Comment:
This is in the dataflow package, so that kinds of Values are in the store isn't known.  So, it's impossible to give that advice here.  But as far as how to implement QualifierHierarchy#widdenUpperBound, I have not advice as it depends on the type system.